templates/cluster/gcp-standalone-cp/templates/k0scontrolplane.yaml

```diff
@@ -0,0 +1,106 @@
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+kind: K0sControlPlane
+metadata:
+  name: {{ include "k0scontrolplane.name" . }}
+spec:
+  replicas: {{ .Values.controlPlaneNumber }}
+  version: {{ .Values.k0s.version }}
+  k0sConfigSpec:
+    args:
+      - --enable-worker
+      - --enable-cloud-provider
+      - --kubelet-extra-args="--cloud-provider=external"
+      - --disable-components=konnectivity-server
+    k0s:
+      apiVersion: k0s.k0sproject.io/v1beta1
+      kind: ClusterConfig
+      metadata:
+        name: k0s
+      spec:
+        api:
+          extraArgs:
+            anonymous-auth: "true"
+          {{- with .Values.k0s.api.extraArgs }}
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+        network:
+          provider: calico
+          calico:
+            mode: ipip
+# TODO: uncomment once https://github.com/Mirantis/helm-charts/pull/3 and
+# https://github.com/Mirantis/helm-charts/pull/4 land
+#        extensions:
+#          helm:
+#            repositories:
+#              - name: mirantis
+#                {{- if .Values.extensions.chartRepository }}
```

Comment:
if one/all of the `values.extensions` nested fields are empty, will the CCM and the CSI charts work as expected?

Reply:
Yes, it should properly work. If `values.extension.chartRepository` is empty, the `mirantis` repository will be configured to `https://charts.mirantis.com` (see the `repositories` section above). If `values.extension.imageRepository`, all the images will be taken from the default repo configured in values.yaml of the CCM and CSI charts: https://github.com/Mirantis/helm-charts/pull/4/files#diff-c48be96ee613c8bab144a07470aaf7ba72a0946ce139959813a50924cc95acd9

Reply:
Right, I have missed the `if` statement hence the question. Now it is all clear to me, thanks for the clarification